### DISCUSSION TITLE
config changes

### DISCUSSION THREAD

**Discussion by AkshayNemade** [eleurent/highway-env#171] [General]
I am trying to run the episodes for 10s with 25 fps.

is changing the config to following values correct thing to do?:
env = gym.make("highway-v0")
env.config['policy_frequency'] = 25
env.config['duration'] = 250

After doing so the env seems to "hand" while running the episodes.

No matter what action we send the vehicle position is just same and doesnt change.
Any insights over this issue will be helpful. 
Thank you :)

**Comment by eleurent** [comment]
Hi @AkshayNemade 
Yes, documentation is still missing on that matter (#62), sorry about that.

You should use:
```
env = gym.make("highway-v0")
env.config['simulation_frequency'] = 25
env.config['policy_frequency'] = 1 ???
env.config['duration'] = 250
env.reset()
```

You could use policy_frequency = simulation_frequency = 25, but that wouldn't make much sense since the discrete actions used (by default) in highway-env (e.g. lane changes) take about 1s to be executed, which is why policy_frequency = 1 by default.
A high-frequency policy makes more sense if you use continuous actions (a 2D space for throttle and steering), which can be achieved by using
```env.config["action"] = {"type": "Continuous"}```

**Comment by AkshayNemade** [comment]
@eleurent  Thank you so much for the prompt reply :) 

My goal was to sample the observation at 25 Hz as we run the episode, hence the policy frequency change. Is there any other way to do that?

Also in regards to continuous actions i am assuming the action input should be a tuple 
eg: `env.step((0.1,0.5))
`
But this creates error in _reward function as it seems to only take discrete actions:
```
\highway_env\envs\highway_env.py in _reward(self, action)
     80             + self.RIGHT_LANE_REWARD * self.vehicle.target_lane_index[2] / (len(neighbours) - 1) \
     81             + self.HIGH_SPEED_REWARD * self.vehicle.speed_index / (self.vehicle.SPEED_COUNT - 1)
---> 82         return utils.remap(action_reward[action] + state_reward,
     83                            [self.config["collision_reward"], self.HIGH_SPEED_REWARD + self.RIGHT_LANE_REWARD],
     84                            [0, 1])

```

Or may be I am doing something wrong...

**Comment by eleurent** [comment]
Yes, to get the observations at 25Hz you need to set the policy_frequency to 25.
But the highway-v0 environment was initially conceived for high-level (low-frequency) discrete actions like maneuvers. If you really wish to make low-level throttle/steering control at 25Hz instead, the reward function must be modified accordingly, as you noticed. I guess the action reward term (used to penalize lane changes) which raises an exception can simply be removed, but other terms must be added, e.g. to ensure that the car does not drive off-road.
